src/metrics/include/metrics/metrics.hpp

```diff
@@ -557,55 +570,84 @@ class Operation {
-                                    this->_counters[name + "_iters"],
-                                    this->_counters[name + "_docs"],
-                                    this->_counters[name + "_bytes"]);
-        return Operation{std::move(op)};
+    OperationT<ClockSource> operation(const std::string& name) {
+        auto op = v1::OperationImpl<ClockSource>(name,
+                                                 this->_timers[name + "_timer"],
+                                                 this->_counters[name + "_iters"],
+                                                 this->_counters[name + "_docs"],
+                                                 this->_counters[name + "_bytes"]);
+        return OperationT{std::move(op)};
     }
 
     // passkey:
-    const std::unordered_map<std::string, v1::CounterImpl>& getCounters(v1::Permission) const {
+    const std::unordered_map<std::string, v1::CounterImpl<ClockSource>>& getCounters(
+        v1::Permission) const {
         return this->_counters;
     };
 
-    const std::unordered_map<std::string, v1::TimerImpl>& getTimers(v1::Permission) const {
+    const std::unordered_map<std::string, v1::TimerImpl<ClockSource>>& getTimers(
+        v1::Permission) const {
         return this->_timers;
     };
 
-    const std::unordered_map<std::string, v1::GaugeImpl>& getGauges(v1::Permission) const {
+    const std::unordered_map<std::string, v1::GaugeImpl<ClockSource>>& getGauges(
+        v1::Permission) const {
         return this->_gauges;
     };
 
-    const time_point now(v1::Permission) const {
-        return metrics::clock::now();
+    const typename ClockSource::time_point now(v1::Permission) const {
+        return ClockSource::now();
     }
 
 private:
-    std::unordered_map<std::string, v1::CounterImpl> _counters;
-    std::unordered_map<std::string, v1::TimerImpl> _timers;
-    std::unordered_map<std::string, v1::GaugeImpl> _gauges;
+    std::unordered_map<std::string, v1::CounterImpl<ClockSource>> _counters;
+    std::unordered_map<std::string, v1::TimerImpl<ClockSource>> _timers;
+    std::unordered_map<std::string, v1::GaugeImpl<ClockSource>> _gauges;
 };
 
+}  // namespace v1
+
+class MetricsClockSource {
+private:
+    using clock_type = std::chrono::steady_clock;
+
+public:
+    using duration = clock_type::duration;
+    using time_point = std::chrono::time_point<clock_type>;
+
+    static time_point now() {
+        return clock_type::now();
+    }
+};
+
+using Registry = v1::RegistryT<MetricsClockSource>;
+
 static_assert(std::is_move_constructible<Registry>::value, "move");
 static_assert(std::is_move_assignable<Registry>::value, "move");
 
+// Registry::clock is actually the ClockSource template parameter, so we check the underlying
+// clock_type in a roundabout way by going through the time_point type it exposes.
+static_assert(Registry::clock::time_point::clock::is_steady, "clock must be steady");
```

Comment:
consider making this a typedef on `Registry`? idk.
```c++
class Registry { public: 
  using clock_is_steady = ClockSource::time_point::clock::is_steady;
};
```

Reply:
`Registry::clock` is the source of truth for how `::now()` is called so my preference is to check against that still. I'm inclined to leave this as-is.